velox/exec/ExchangeSource.h

```diff
@@ -86,8 +86,14 @@ class ExchangeSource : public std::enable_shared_from_this<ExchangeSource> {
   /// backward compatibility (e.g. communicating with coordinator), we allow
   /// small data (1MB) to be returned.
   virtual folly::SemiFuture<Response> requestDataSizes(
-      uint32_t /*maxWaitSeconds*/) {
-    VELOX_NYI();
+      uint32_t maxWaitSeconds) = 0;
+
+  const std::vector<int64_t>& remainingBytes() const {
+    return remainingBytes_;
+  }
+
+  void setRemainingBytes(std::vector<int64_t> remainingBytes) {
```

Comment:
It feels a bit hacky to have the 'source' return this information as part of requestDataSizes and also have the caller use this method to set it "back". It might be cleaner for the caller to store it elsewhere or have the source store it on its own.

Also, should this be called while holding queue lock? If so, the method name needs to have 'Locked' suffix.

BTW, can we add checks somewhere to ensure that all entries in the arrays are > 0?